### PR TITLE
cleanup markdown

### PR DESCRIPTION
**Pull request by ilrudie**
cleanup some linter suggestions for our markdown

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -49,4 +49,3 @@ It’s important that every piece of code in K8sGateway is reviewed by at least
 3. **Testing**: Please write tests for your changes. Bias towards fast / unit testing.
 4. **Comments**: The code reviewer may leave comments to discuss changes. Minor preferences are often called out with `nit`.
 5. **PR Description**: Is well formatted, concise and descriptive enough to be used as a release note.
-
```
Comment (craigbox) [inline CONTRIBUTING.md:52]: the Istio linter normally [prefers a new line at the end of markdown files](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md047---files-should-end-with-a-single-newline-character).  Is it possible to have the same rules for both?

Comment (ilrudie) [reply CONTRIBUTING.md]: we should do this, yeah. the lack of newline is a mistake on my part

Comment (ilrudie) [reply CONTRIBUTING.md]: oh, some oddball thing with editing the file in vscode (which I don't normally use) strips 1 newline from the end of the file on save 🤷‍♂️ I got nothing.

**Review by jenshu — APPROVED** [review]
the formatting LGTM

**Pull request merged**